source/common/upstream/cluster_manager_impl.cc

```diff
@@ -417,7 +417,13 @@ void ClusterManagerImpl::onClusterInit(Cluster& cluster) {
   // been setup for cross-thread updates to avoid needless updates during initialization. The order
   // of operations here is important. We start by initializing the thread aware load balancer if
   // needed. This must happen first so cluster updates are heard first by the load balancer.
-  auto cluster_data = active_clusters_.find(cluster.info()->name());
+  auto cluster_data = warming_clusters_.find(cluster.info()->name());
+  if (cluster_data != warming_clusters_.end()) {
```

Comment:
Isn't the point of this change that when this function is called the cluster is always in warming? Can you add more comments above to clarify when the cluster would not be in warming?

Reply:
The role of this change is to assure clusters which called this function should be active. Comment addition done.

Reply:
My question is how can it be that a cluster is not warming when this function is called?

/wait-any

Reply:
[this](https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/cluster_manager_impl.cc#L294) section will be immediately active.

Reply:
OK please add more comments.

/wait

Reply:
comment addition done